### PR TITLE
Avoid RedisCacheStore#increment on Rails 6+

### PR DESCRIPTION
**Pull request by jdelStrother**
As far as I can tell, Rails has handled increment+expires_in since
6.0 (https://github.com/rails/rails/commit/9d5b02ec5062a23665ec596ef7d3efe4f5abcc27), and more recent versions add pipeline optimizations on top of that.

How about avoiding our own custom RedisCacheStoreProxy#increment method on modern versions of ActiveSupport?

This also avoids an extra GET (from the `if options[:expires_in] && !read(name)` call).

### CODE REVIEW

**Comment by jdelStrother** [comment]
Let me know if there's interest in merging this & I'll rebase.

**Comment by mitchellhenke** [comment]
I'd really like to see it land.

In some brief local application-level benchmarking, the patch is quite a bit faster. In simpler applications or situations where Redis latency is worse, I suspect the impact would be even larger. Using [derailed_benchmarks](https://github.com/zombocom/derailed_benchmarks):

```
❤️ ❤️ ❤️  (Statistically Significant) ❤️ ❤️ ❤️

(0.2698 seconds) "rack attack update"
  FASTER 🚀🚀🚀 by:
    1.0643x [older/newer]
    6.0431% [(older - newer) / older * 100]
(0.2872 seconds) "CONSTANT"

Iterations per sample: 200
Samples: 200

Test type: Kolmogorov Smirnov
Confidence level: 99.0 %
Is significant? (max > critical): true
D critical: 0.15174271293851463
D max: 0.66
```

**Comment by grzuy** [comment]
> Let me know if there's interest in merging this & I'll rebase.

Yes, please.

Thank you.

**Review by grzuy — APPROVED** [review]
Makes sense!

Thank you!

Pending conflicts with latest changes in main.

**Comment by jdelStrother** [comment]
I'm away for a few weeks, but can pick this up again in November.

File: lib/rack/attack/store_proxy/redis_cache_store_proxy.rb [hunk 1]
```diff
@@ -10,17 +10,19 @@ def self.handle?(store)
...
         end
 
-        def increment(name, amount = 1, **options)
-          # RedisCacheStore#increment ignores options[:expires_in].
-          #
-          # So in order to workaround this we use RedisCacheStore#write (which sets expiration) to initialize
-          # the counter. After that we continue using the original RedisCacheStore#increment.
-          if options[:expires_in] && !read(name)
-            write(name, amount, options)
+        if defined?(ActiveSupport) && ActiveSupport::VERSION::MAJOR < 6
```
Comment (santib) [inline lib/rack/attack/store_proxy/redis_cache_store_proxy.rb:13]:
```suggestion
        if defined?(::ActiveSupport) && ::ActiveSupport::VERSION::MAJOR < 6
```

Comment (santib) [inline lib/rack/attack/store_proxy/redis_cache_store_proxy.rb:13]: I'd suggest adding a gemfile that runs `spec/acceptance/stores/active_support_redis_cache_store_spec.rb` with activesupport < 6

Comment (grzuy) [reply lib/rack/attack/store_proxy/redis_cache_store_proxy.rb:13]: Separate PR?

Comment (santib) [reply lib/rack/attack/store_proxy/redis_cache_store_proxy.rb:13]: Done https://github.com/rack/rack-attack/pull/632

**Comment by mitchellhenke** [comment]
I'm very interested in seeing this merged, is there anything I can do to help move it forward?

**Comment by santib** [comment]
Hi @mitchellhenke, I think we are waiting for @jdelStrother to fix the CI and that should be it 👌

He said he was out til November, so he'll probably be coming back soon.

**Comment by jdelStrother** [comment]
Yep, let me try & take a look at it today

**Comment by jdelStrother** [comment]
Rebased and fixed the `defined?(ActiveSupport)` check, though Github Actions seems to be having a bad day - could you re-run the failed ones?

**Comment by santib** [comment]
Thanks for your contribution @jdelStrother 🙌

**Pull request merged**